src/HDInsight/HDInsight/Models/Management/AzureHdInsightManagementClient.cs

```diff
@@ -225,5 +226,21 @@ public virtual void RestartHosts(string resourceGroupName, string clusterName, I
         {
             HdInsightManagementClient.VirtualMachines.RestartHosts(resourceGroupName, clusterName, hosts);
         }
+        
+        private void ResetClusterIdentity(ClusterCreateParametersExtended createParams, string cloudAadAuthority, string cloudDataLakeAudience)
+        {
+            var configuation = (Dictionary<string, Dictionary<string, string>>)createParams.Properties.ClusterDefinition.Configurations;
+            Dictionary<string, string> clusterIdentity;
+            if(!configuation.TryGetValue("clusterIdentity", out clusterIdentity))
+            {
+                return;
+            }
+            clusterIdentity.SetProperty("clusterIdentity.resourceUri", cloudDataLakeAudience);
+
+            string aadTenantIdWithUrl;
+            clusterIdentity.TryGetValue("clusterIdentity.aadTenantId", out aadTenantIdWithUrl);
+            string newAadTenantIdWithUrl = aadTenantIdWithUrl?.Replace("https://login.windows.net/", cloudAadAuthority);
```

Comment:
Nit - nice to extract a constant for "https://login.windows.net/"

Reply:
I agree with Dongwei.

Reply:
> Nit - nice to extract a constant for "https://login.windows.net/"

Fixed.